lib/kafka/server.go

```diff
@@ -0,0 +1,108 @@
+package kafka
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/zorkian/kafka"
+	"github.com/discordapp/punt/lib/syslog"
+)
+
+type Server struct {
+	Messages chan syslog.SyslogData
+	Broker* kafka.Broker
+	FreightTopic string
+}
+
+type ServerConfig struct {
+	FreightTopic string 
+	Address string
+}
+
+type AudioMetrics struct {
+	Timestamp int64 `json:"timestamp"`
+	MediaSessionID string `json:"media_session_id"`
+	User struct {
+		Rtt interface{} `json:"rtt"`
+		Protocol interface{} `json:"protocol"`
+		ID string `json:"id"`
+		AudioInbound struct {
+			PacketsLost int `json:"packets_lost"`
+			Ssrc int `json:"ssrc"`
+			Jitter int `json:"jitter"`
+			PacketsReceived int `json:"packets_received"`
+			AudioLevel float64 `json:"audio_level"`
+		} `json:"audio_inbound"`
+		AudioOutbound struct {
+			PacketsSent int `json:"packetsSent"`
+			Ssrc int `json:"ssrc"`
+			AudioLevel float64 `json:"audioLevel"`
+		} `json:"audioOutbound"`
+	} `json:"user"`
+	Channel struct {
+		ID string `json:"id"`
+	} `json:"channel"`
+	Server struct {
+		Hostname string `json:"hostname"`
+		ID string `json:"id"`
+	} `json:"server"`
+}
+
+func NewServer(config ServerConfig, messages chan syslog.SyslogData) *Server {
+
+	broker, err := kafka.NewBroker("kafka-default", []string{config.Address}, kafka.NewBrokerConf("discord-clerk"))
+	if err != nil {
+		panic(err)
+	}
+
+	server := Server{
+		Messages: messages,
+		FreightTopic: config.FreightTopic,
+		Broker: broker,
+	}
+	return &server
+}
+
+func (s *Server) Start() {
+	partitions, err := s.Broker.PartitionCount(s.FreightTopic)
+	if err != nil {
+		panic(err)
+	}
+
+	for partitionID := int32(0); partitionID < partitions; partitionID++ {
+		go s.partitionReader(s.FreightTopic, partitionID)
+	}
+}
+
+func (s* Server) partitionReader(topic string, partitionID int32) {
+	conf := kafka.NewConsumerConf(topic, partitionID)
+	consumer, err := s.Broker.Consumer(conf)
+	if err != nil {
+		panic(err)
+	}
+	for {
+		msg, err := consumer.Consume()
+		if err != nil {
+			panic(err)
+		}
+
+		unpack := make(map[string]interface{})
+		if err := json.Unmarshal(msg.Value, &unpack); err != nil {
+			panic(err)
+		}
+		var audio_metrics AudioMetrics
```

Comment:
I think for now what you want to do is to just assume that the payload we get from kafka has a 'timestamp' field. So instead of a `AudioMetrics` struct, I would just have a `Timestamp` struct with one field `Timestamp` and use that here to parse the timestamp.

In the future I can clean this up to support passing a timestamp key into the kafka ingestor configuration.

Reply:
True, I only need the timestamp.

Reply:
Done.